ipa-core/src/cli/crypto/encrypt.rs

```diff
@@ -0,0 +1,269 @@
+            "53d58e022981f2edbf55fec1b45dbabd08a3442cb7b7c598839de5d7a5888bff",
+            "3a0a993a3cfc7e8d381addac586f37de50c2a14b1a6356d71e94ca2afaeb2569",
+            "1fb5c5274bf85fbe6c7935684ef05499f6cfb89ac21640c28330135cc0e8a0f7",
+        ];
+
+        #[allow(clippy::large_futures)]
+        let results = join3v(
+            EncryptedOprfReportStreams::from(files)
+                .streams
+                .into_iter()
+                .zip(world.contexts())
+                .zip(mk_private_keys.into_iter())
+                .map(|((input, ctx), mk_private_key)| {
+                    let mk_private_key = hex::decode(mk_private_key)
+                        .map(|bytes| IpaPrivateKey::from_bytes(&bytes).unwrap())
+                        .unwrap();
+                    let query_config = IpaQueryConfig {
+                        max_breakdown_key: 3,
+                        with_dp: 0,
+                        epsilon: 1.0,
+                        ..Default::default()
+                    };
+
+                    OprfIpaQuery::<_, BA16, _>::new(
+                        query_config,
+                        Arc::new(KeyRegistry::from_keys([PrivateKeyOnly(mk_private_key)])),
+                    )
+                    .execute(ctx, query_size, input)
+                }),
+        )
+        .await;
+
+        assert_eq!(
+            results.reconstruct()[0..3]
+                .iter()
+                .map(U128Conversions::as_u128)
+                .collect::<Vec<u128>>(),
+            EXPECTED
+        );
+    }
+
+    #[test]
+    #[should_panic = "Failed to open network file:"]
+    fn encrypt_no_network_file() {
+        let input_file = sample_data::write_csv(sample_data::test_ipa_data().take(10)).unwrap();
+
+        let output_dir = tempdir().unwrap();
+        let network_dir = tempdir().unwrap();
+        let network_file = network_dir.path().join("does_not_exist");
+        EncryptArgs::new(input_file.path(), output_dir.path(), &network_file)
+            .encrypt()
+            .unwrap();
+    }
+
+    #[test]
+    #[should_panic = "TOML parse error at"]
+    fn encrypt_bad_network_file() {
+        let input_file = sample_data::write_csv(sample_data::test_ipa_data().take(10)).unwrap();
+        let output_dir = tempdir().unwrap();
+        let network_data = r"
+this is not toml!
+%^& weird characters
+(\deadbeef>?
+";
+        let mut network_file = NamedTempFile::new().unwrap();
+        writeln!(network_file.as_file_mut(), "{network_data}").unwrap();
+
+        EncryptArgs::new(input_file.path(), output_dir.path(), network_file.path())
+            .encrypt()
+            .unwrap();
+    }
+
+    #[test]
+    #[should_panic = "invalid length 2, expected an array of length 3"]
+    fn encrypt_incomplete_network_file() {
+        let input_file = sample_data::write_csv(sample_data::test_ipa_data().take(10)).unwrap();
+
+        let output_dir = tempdir().unwrap();
+        let network_data = r#"
```

Comment:
should this use the `format!` macro used below? I didn't know that existed when I wrote this

Reply:
we could use it, but I thought for this test we don't need to inject the correct keys, so the benefit of using `format!` is not clear